capstone/src/main/webapp/js/autocomplete.js

```diff
@@ -0,0 +1,167 @@
+/*
+ * Largely inspired by JS code from W3Schools: https://www.w3schools.com/howto/howto_js_autocomplete.asp
+ */
+const SEARCH_INPUT = document.getElementById('search');
+const ITEM_CLASS = 'autocomplete-items';
+const ACTIVE_CLASS = 'autocomplete-active';
+const LIST_ID = 'autocomplete-list';
+const UP_KEY = 40;
+const DOWN_KEY = 38;
+const ENTER = 13;
+let selectedElmntPos = -1;
+
+/**
+ * Adds the 'active' tag to an autocomplete elmt
+ * @param {Object} autocompleteListElmt
+ * @return {bool}
+ */
+function addActiveTag(autocompleteListElmt) {
+  if (!autocompleteListElmt) {
+    return false;
+  }
+  removeActiveTag(autocompleteListElmt);
+  if (selectedElmntPos >= autocompleteListElmt.length) {
+    selectedElmntPos = 0;
+  } else if (selectedElmntPos < 0) {
+    selectedElmntPos = (autocompleteListElmt.length - 1);
+  }
+  autocompleteListElmt[selectedElmntPos].classList.add(ACTIVE_CLASS);
+}
+
+/**
+ * Removes the 'active' tag from an autocomplete elmt
+ * @param {Object} autocompleteListElmt
+ */
+function removeActiveTag(autocompleteListElmt) {
+  for (elmt of autocompleteListElmt) {
+    elmt.classList.remove(ACTIVE_CLASS);
+  }
+}
+
+/**
+ * Closes dropdown autocomplete list
+ * Does not close the element provided
+ * @param {Object} elmnt
+ * @return {void}
+ */
+function closeAllElmntExcept(elmnt) {
+  const autoItems = document.getElementsByClassName(ITEM_CLASS);
+  for (const item of autoItems) {
+    if (item != elmnt && item != SEARCH_INPUT) {
+      item.parentNode.removeChild(item);
+    }
+  }
+}
+
+/**
+ * Adds valid suggestions to list of stored suggestions
+ * @param{string} validSuggestion
+ */
+function keepTrackOfSuggestions(validSuggestion) {
+  const parent = document.getElementById('suggestions');
+  const listElt = document.createElement('li');
+  listElt.innerHTML = validSuggestion;
+  parent.appendChild(listElt);
+}
+
+/**
+ * Current placeholder until servlet is created
+ * @return{Array<string>} arr
+ */
+function getListOfSuggestions() {
```

Comment:
In reality this would be an async method. So make sure your rendering logic still works with a real server too. No actions here. Just a heads up.